src/unix/notbsd/linux/other/mod.rs

```diff
@@ -1,5 +1,18 @@
 pub type __priority_which_t = ::c_uint;
 
+s_no_extra_traits! {
+    pub union XXX_not_sure_what_to_call_this {
+        addr_bnd: addr_bnd_t,
+        pkey: u32,
+    }
+
+    #[repr(C, packed)]
+    pub union sifields_t {
```

Comment:
Why is this union `packed` ? 

cc @joshtriplet why do we accept `repr(packed)` on `union`s at all ? (it is a concept that IMO does not make much sense, at least for `repr(C)` unions, which are always packed.

Reply:
https://github.com/rust-lang/libc/issues/716#issuecomment-331675843

Reply:
cc @joshtriplett, @rkruppe: why do we accept `repr(packed)` on `union`s at all ? Is that something that makes sense? AFAICT for `repr(C)` unions at least it does not, they are "always" packed (all fields start at offset 0, there is no padding, etc.).

Reply:
`repr(packed)` also lowers the alignment requirement of the annotated type (`repr(align(N))` can only raise it), which is also useful for unions and -- from a cursory glance at the linked issue -- apparently exactly what's needed here.

Reply:
> repr(packed) also lowers the alignment requirement of the annotated type 

Ah, that makes sense then. When used on `union`s, is it then equivalent to `repr(align(1))` ? 

---

I thought that it lowered the alignment requirements of the fields, which impacted how the alignment of the annotated was computed.

Reply:
> I thought that it lowered the alignment requirements of the fields, which impacted how the alignment of the annotated was computed.

You can also view it that way if you want, then lowering the alignment of the whole type is a consequence that also applies to unions (since the requirements of the field(s) at offset 0 also matters). `repr(align(1))` won't help, `repr(align)` can only *raise* alignment.